libraries/wasm-jit/Include/IR/Operators.h

```diff
@@ -675,4 +675,19 @@ namespace IR
 	};
 
 	IR_API const char* getOpcodeName(Opcode opcode);
-}
\ No newline at end of file
+}
+
+//paranoia for future platforms
+static_assert(sizeof(IR::OpcodeAndImm<IR::NoImm>) == 2);
+static_assert(sizeof(IR::OpcodeAndImm<IR::MemoryImm>) == 2);
+static_assert(sizeof(IR::OpcodeAndImm<IR::ControlStructureImm>) == 3);
+static_assert(sizeof(IR::OpcodeAndImm<IR::BranchImm>) == 6);
+static_assert(sizeof(IR::OpcodeAndImm<IR::BranchTableImm>) == 18);
+static_assert(sizeof(IR::OpcodeAndImm<IR::LiteralImm<I32>>) == 6);
+static_assert(sizeof(IR::OpcodeAndImm<IR::LiteralImm<I64>>) == 10);
+static_assert(sizeof(IR::OpcodeAndImm<IR::LiteralImm<F32>>) == 6);
+static_assert(sizeof(IR::OpcodeAndImm<IR::LiteralImm<F64>>) == 10);
```

Comment:
Do we want to protect `LiteralImm` will not be instantiated by types other than I32, I64, F32, F64?

Reply:
I think it's really only used here?
https://github.com/AntelopeIO/leap/blob/bef672aae24a061b8b10b30ca6de6f4cf7718320/libraries/wasm-jit/Include/IR/Operators.h#L157-L160
(the SIMD types will never be used; and possibly that SIMD stuff should be ripped out due to it being dead cruft)

Even so, I wasn't necessarily trying to be exhaustive. tbh the very thorough wasm spec tests should shake out problems in any of these. Just nice to catch the problem early (during compile), and, for the `OpcodeAndImm<ControlStructureImm>` case, make it super ultra clear the change in consensus code was the same before and after.

Reply:
Thanks for the explanation!

Reply:
Yes that SIMD code should be taken out.